libs/wingc/src/lib.rs

```diff
@@ -253,6 +255,53 @@ fn add_builtin(name: &str, typ: Type, scope: &mut Scope, types: &mut Types) {
 		.expect("Failed to add builtin");
 }
 
+struct CompilationContext {
+	/// Description of current compilation phase
+	phase: String,
+
+	/// Location in source we're currently processing
+	span: WingSpan,
+}
+
+static mut COMPILATION_CONTEXT: Option<CompilationContext> = None;
```

Comment:
Any reason to not use the same pattern as the lsp? (`thread_local!` with RefCell)

Reply:
Not really, kind of looked at this and figured to go as simple as possible and build up from there:
![image](https://github.com/winglang/wing/assets/1160578/e919442c-ff44-4ace-91a6-2a74888899cd)